### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by lincarl1**
Changed the admin page so that it looks more similar to our wireframe, fixed a bug where I was checking for display name instead of author id to delete posts and comments, added the kick functionality.

### CODE REVIEW

**Comment by lincarl1** [comment]
Screen Shot 2021-04-13 at 3 41 29 PM
Screen Shot 2021-04-13 at 3 40 52 PM

**Review by rieckn — APPROVED** [review]
Looks Good!

File: src/AppContext.js [hunk 1]
```diff
@@ -56,6 +57,13 @@ const reducer = produce((draft, action) => {
             draft.joinSuccess = action.join.joinSuccess
             draft.discussionName = action.join.discussionName
             break
+        case 'kicked':
+            draft.kicked.push(action.kicked)
```
Comment (ngynkvn) [inline src/AppContext.js:61]: Why does this need to be included client-side?

File: src/Views/EnterDisplayName.js [hunk 2]
```diff
@@ -12,28 +13,68 @@ export default function EnterDisplayName(props) {
...
-                        required
-                    />{' '}
+    function handleRejoinClick(e) {
+        e.preventDefault()
+        API.join(name, props.match.params.roomID)
+        dispatch({ type: 'update-name', displayName: name})
+        dispatch({ type: 'remove-from-kicked', userId: contextState.userId})
+    }
+
+    if(contextState.kicked.includes(contextState.userId) === true)
```
Comment (ngynkvn) [inline src/Views/EnterDisplayName.js:23]: This is superfluous. We don't need an entire array if the entire point is to check if a single user has been kicked. I'd rather a boolean value here in the application state than an array.

File: src/Views/Room.js [hunk 3]
```diff
@@ -19,12 +19,27 @@ function Room(props) {
         }
     }, [])
 
-    return (
-        <div className="room-page">
-            {contextState.joinSuccess && <Header roomKey={contextState.roomKey} />}
-            {contextState.displayName !== null && contextState.joinSuccess ? <RoomHomeScreen /> : <EnterDisplayName {...props} />}
-        </div>
-    )
+    if(contextState.kicked.includes(contextState.userId))
```
Comment (ngynkvn) [inline src/Views/Room.js:22]: I don't like this, dispatch calls don't belong here.

File: src/Views/EnterDisplayName.js (hunk 2 above)
Comment (ngynkvn) [reply src/Views/EnterDisplayName.js:23]: We are also branching for no reason. The code paths are nearly the same.

File: src/AppContext.js [hunk 4]
```diff
@@ -161,6 +172,12 @@ const socketEvents = (dispatch) => {
         //     handleClose();
         // console.log(contextState.posts)
     })
+    socket.on('kicked', (userId) => {
+        dispatch({type: 'kicked', kicked: userId })
+        // dispatch({type: 'join-successful', joinSuccess: false })
```
Comment (ngynkvn) [inline src/AppContext.js:177]: Why not have the dispatch calls here?

**Review by ngynkvn — CHANGES_REQUESTED** [review]

File: src/Views/EnterDisplayName.js (hunk 2 above)
Comment (lincarl1) [reply src/Views/EnterDisplayName.js:23]: If we are sending a boolean with the userId that is being kicked, how would I reference that specific userId?

Comment (lincarl1) [reply src/Views/EnterDisplayName.js:23]: I agree that the boolean would be easier

Comment (lincarl1) [reply src/Views/EnterDisplayName.js:23]: Would it be with the draft case?

**Pull request merged**